src/index.ts

```diff
@@ -87,19 +99,36 @@ export default function (
 					[...usedIcons, ...(await getExtraIcons(options))],
 					options,
 				);
-				const file = path.join(dir.output, options.sprite.writeFile as string);
-				const fileDirectory = path.parse(file).dir;
+
+				const relFileUrl = await getFileRelativeUrl(options);
+				assert(typeof relFileUrl === 'string', 'Unexpected type of relFileUrl');
+
+				const outputFilepath = path.join(dir.output, relFileUrl);
+
+				const fileDirectory = path.parse(outputFilepath).dir;
 				try {
 					await fs.readdir(fileDirectory);
 				} catch {
 					await fs.mkdir(fileDirectory, { recursive: true });
 				}
-				await fs.writeFile(file, sprite);
+				await fs.writeFile(outputFilepath, sprite);
 			},
 		);
 	}
 
 	for (const source of options.sources) {
 		eleventyConfig.addWatchTarget(source.path);
 	}
+
+	const getFileRelativeUrl = async (
+		opts: Options,
+	): Promise<string | undefined> => {
+		if (opts.sprite.writeFile !== false) {
+			return pathToUrl(path.join(opts.sprite.writeFile as string));
+		}
+
+		return undefined;
+	};
+
+	const pathToUrl = (pathStr: string) => pathStr.split(path.sep).join('/');
```

Comment:
```suggestion
	const pathToUrl = (p: string) => p.split(path.sep).join('/');
```

Sorry, variable naming nit.